tests/storage/restore.go

```diff
@@ -907,8 +954,23 @@ var _ = SIGDescribe("VirtualMachineRestore Tests", func() {
 						&expect.BExp{R: console.RetValue("0")},
 						&expect.BSnd{S: catTestDataMessageCmd},
 						&expect.BExp{R: string(vm.UID)},
+						&expect.BSnd{S: console.EchoLastReturnValue},
+						&expect.BExp{R: console.RetValue("0")},
 					}...)
 
+					if tpm {
+						batch = append(batch, []expect.Batcher{
+							&expect.BSnd{S: fmt.Sprintf("sudo tpm2_nvread -s %d -C o 1\n", len(string(vm.UID)))},
```

Comment:
I think the VMI needs to be post-firmware for these assertions - https://github.com/kubevirt/kubevirt/pull/12711
Long shot but maybe this is why it flakes?

Reply:
Somewhat discussed offline, but also wanted to add that this is using `LoginToFedora` which I believe is safe and isn't related to the flake.

Reply:
Actually, and I know this is disgusting, if you sent a keystroke accidentally to the firmware, it's going to break
https://github.com/kubevirt/kubevirt/pull/12711
If you could just add this small check into the test :pray:

Reply:
Okay will do. Unrelated to this comment but wdyt about returning error to force requeue only in the restore controller as suggested in https://github.com/kubevirt/kubevirt/pull/13207#discussion_r1909071834?

Reply:
Done